x-pack/platform/packages/shared/kbn-sample-parser/client/parse_dataset.ts

```diff
@@ -17,11 +17,26 @@ export function parseDataset({ system, parser }: { parser: LoghubParser; system:
       message: line,
     };
   });
-  const sortedLogLines = orderBy(parsedLogLines, (line) => line.timestamp, 'asc');
-  const min = sortedLogLines[0].timestamp;
-  const max = sortedLogLines[parsedLogLines.length - 1].timestamp;
 
-  const count = sortedLogLines.length;
+  const min = parsedLogLines[0].timestamp;
+
+  let minTimestamp = min;
+  let years = 0;
+
+  // add years for timestamps without years
+  parsedLogLines.forEach((logLine) => {
+    if (logLine.timestamp < minTimestamp) {
+      minTimestamp = logLine.timestamp;
+      years++;
```

Comment:
Every time a timestamp is smaller than the one of the current minimum, you increment a counter and that's the number of years you are adding to every single document going forward?

Why is it guaranteed that this is what's happening when the timestamp gets smaller? Shouldn't we check whether the year is set or not and react to that?

I'm probably dense but I don't understand what's going on here

Reply:
I'll add a comment. Basically if there is no year in the timestamp (it only has month-day-hour etc), the parser will create a date with the current year. The lines are already sorted by time of occurrence. The Proxifier sample logs look like this:

```
[10.30 21:21:48] ...
[07.26 13:30:34] ...
```

if you create timestamps with the same year for both they actually end up being in the wrong order. So if the current timestamp is smaller than the previous timestamp, we assume it's because it's skipped a year.

That being said, HPC has a different issue, because that one has epoch timestamps. I'll look into it

Reply:
Ah, the part about it adding the current year was the missing bit, makes sense.

Reply:
the timestamps for HPC were simply not sorted correctly